### PR TITLE
CORTX-29612: CORTX-29642: Confstore: alternate PR for PR #814 (DCO check issue)

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

Raised this PR for PR:https://github.com/Seagate/cortx-utils/pull/814
Above PR had DCO failures with good number of commits hence had raised this PR with all the changes present in PR: #814

**This PR includes changes from following PRs:**
https://github.com/Seagate/cortx-utils/pull/749
https://github.com/Seagate/cortx-utils/pull/746
https://github.com/Seagate/cortx-utils/pull/755
https://github.com/Seagate/cortx-utils/pull/756
https://github.com/Seagate/cortx-utils/pull/763
https://github.com/Seagate/cortx-utils/pull/779
https://github.com/Seagate/cortx-utils/pull/780

Relevant Stories:

[CORTX-29612](https://jts.seagate.com/browse/CORTX-29612): ConfStore: instead of None, use empty string as a null value
[CORTX-29642](https://jts.seagate.com/browse/CORTX-29642): Conf_store: After deletetion of keys from conf, Conf.get_keys() still returns the deleted keys.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]:
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]:
- Confirm that Test Cases are added for new features added [Y/N]: 
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  

# Review Checklist 
####  Before posting the PR please ensure:
- [ ] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [ ] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [x] ConfStore
- [x] KVStore
- [x] ConfCli
- [x] Test cases added for all above 3
- [x] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Comment by cdeshmukh** [comment]
DCO check doesn't need code changes, but proper signed commits in PR. What are these changes about ?

**Comment by gauravchaudhari02** [comment]
retest this please

**Comment by rahul-tripathi-git** [comment]
Codacy Issue below needs to be ignored and will be suppressed in near future.
Also this is not inline with Coding standards followed by cortx-py-utils.

1 blank line required before class docstring (found 0) (D203)

**Comment by rahul-tripathi-git** [comment]
> DCO check doesn't need code changes, but proper signed commits in PR. What are these changes about ?

Raised this PR for PR:https://github.com/Seagate/cortx-utils/pull/814
Above PR had DCO failures with good number of commits hence had raised this PR with all the changes present in PR: https://github.com/Seagate/cortx-utils/pull/814

CC: @cdeshmukh

**Review by suryakumar1024 — APPROVED** [review]

**Review by rohit-k-dwivedi — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_payload.py [hunk 1]
```diff
@@ -119,8 +119,6 @@ def get_keys(self, starts_with: str = '', recurse: bool = True, **filters) -> li
                     when True, returns keys including array index
                     e.g. In case of "xxx[0],xxx[1]", only "xxx" is returned
         """
-        if all([len(filters.items()) == 0, not starts_with]):
```
[the remark this answers is not in the document]
Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_payload.py:122]: This was a fix for a bug for deletion was not happening with get keys.
https://github.com/Seagate/cortx-utils/pull/779/files

File: py-utils/src/utils/kv_store/kv_payload.py [hunk 2]
```diff
@@ -191,7 +189,7 @@ def _set(self, key: str, val: str, data: dict):
                 data[k[0]] = []
             # if index is more than list size, extend the list
             for i in range(len(data[k[0]]), index + 1):
-                data[k[0]].append({})
+                data[k[0]].append('')
```
[the remark this answers is not in the document]
Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_payload.py:192]: These changes were done to enable strings to be set as value in confstore. It was more of a standardizing measure in conf store.
https://github.com/Seagate/cortx-utils/pull/780/files

**Review by cdeshmukh — APPROVED** [review]
Confirm jenkin jobs details for build, deployment & sanity test results

**Comment by cdeshmukh** [comment]
Codacy scans running for couple of hours now, it has some issues across repos it seems, not waiting for it anymore

**Pull request merged**